File: 백준/Silver/18258. 큐 2/큐 2.py
import sys
from collections import deque

N = int(sys.stdin.readline().strip())
q = deque()

for _ in range(N):
    com = sys.stdin.readline().strip().split()
    if com[0] == 'push':
        q.append(int(com[1]))
    elif com[0] == 'pop':
        if q:
            print(q.popleft())
        else:
            print(-1)
    elif com[0] == 'size':
        print(len(q))
    elif com[0] == 'empty':
        if q:
            print(0)
        else:
            print(1)
    elif com[0] == 'front':
        if q:
            print(q[0])
        else:
            print(-1)
    elif com[0] == 'back':
        if q:
            print(q[-1])
        else:
            print(-1)

File: 백준/Silver/18258. 큐 2/큐 2.cc
#include <iostream>
#include <queue>
#include <sstream>
#include <string>

using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    ostringstream oss;
    int N;

    cin >> N;
    queue<int> q;
    for (int i = 0; i < N; i++)
    {
        string cmd;
        cin >> cmd;

        if (cmd == "push")
        {
            int X;
            cin >> X;
            q.push(X);
        }
        else if (cmd == "pop")
        {
            if (q.empty())
                oss << -1 << "\n";
            else
            {
                oss << q.front() << "\n";
                q.pop();
            }
        }
        else if (cmd == "size")
        {
            oss << q.size() << "\n";
        }
        else if (cmd == "empty")
        {
            oss << (int)q.empty() << "\n";
        }
        else if (cmd == "front")
        {
            if (q.empty())
                oss << -1 << "\n";
            else
                oss << q.front() << "\n";
        }
        else if (cmd == "back")
        {
            if (q.empty())
                oss << -1 << "\n";
            else
                oss << q.back() << "\n";
        }
    }
    cout << oss.str();
}